src/System.Numerics.Matrices/src/System/Numerics/Matrix4x4.cs

```diff
@@ -0,0 +1,435 @@
+
+                fixed (Matrix4x4* p = &this)
+                {
+                    double* d = (double*)p;
+                    d[row * ColumnCount + col] = value;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets the number of columns in the matrix
+        /// </summary>
+        public int Columns { get { return ColumnCount; } }
+        /// <summary>
+        /// Get the number of rows in the matrix
+        /// </summary>
+        public int Rows { get { return RowCount; } }
+
+        /// <summary>
+        /// Gets a new Matrix1x4 containing the values of column 1
+        /// </summary>
+        public Matrix1x4 Column1 { get { return new Matrix1x4(M11, M12, M13, M14); } }
+        /// <summary>
+        /// Gets a new Matrix1x4 containing the values of column 2
+        /// </summary>
+        public Matrix1x4 Column2 { get { return new Matrix1x4(M21, M22, M23, M24); } }
+        /// <summary>
+        /// Gets a new Matrix1x4 containing the values of column 3
+        /// </summary>
+        public Matrix1x4 Column3 { get { return new Matrix1x4(M31, M32, M33, M34); } }
+        /// <summary>
+        /// Gets a new Matrix1x4 containing the values of column 4
+        /// </summary>
+        public Matrix1x4 Column4 { get { return new Matrix1x4(M41, M42, M43, M44); } }
+        /// <summary>
+        /// Gets a new Matrix4x1 containing the values of column 1
+        /// </summary>
+        public Matrix4x1 Row1 { get { return new Matrix4x1(M11, M21, M31, M41); } }
+        /// <summary>
+        /// Gets a new Matrix4x1 containing the values of column 2
+        /// </summary>
+        public Matrix4x1 Row2 { get { return new Matrix4x1(M12, M22, M32, M42); } }
+        /// <summary>
+        /// Gets a new Matrix4x1 containing the values of column 3
+        /// </summary>
+        public Matrix4x1 Row3 { get { return new Matrix4x1(M13, M23, M33, M43); } }
+        /// <summary>
+        /// Gets a new Matrix4x1 containing the values of column 4
+        /// </summary>
+        public Matrix4x1 Row4 { get { return new Matrix4x1(M14, M24, M34, M44); } }
+
+        public override bool Equals(object obj)
+        {
+            if (obj is Matrix4x4)
+                return this == (Matrix4x4)obj;
+
+            return false;
+        }
+
+        public bool Equals(Matrix4x4 other)
+        {
+            return this == other;
+        }
+
+        public unsafe override int GetHashCode()
+        {
+            fixed (Matrix4x4* p = &this)
+            {
+                int* x = (int*)p;
+                unchecked
+                {
+                    return 0xFFE1
+                         + 7 * ((((x[00] ^ x[01]) << 0) + ((x[02] ^ x[03]) << 1) + ((x[04] ^ x[05]) << 2) + ((x[06] ^ x[07]) << 3)) << 0)
+                         + 7 * ((((x[04] ^ x[05]) << 0) + ((x[06] ^ x[07]) << 1) + ((x[08] ^ x[09]) << 2) + ((x[10] ^ x[11]) << 3)) << 1)
```

Comment:
This logic could be hokum, somebody with knowledge of hashing should take a closer look. This kind of logic has always worked finely for me in the past, but it very well could be over designed and/or a terrible algo.